### PR TITLE
Rust-ify

### PR DESCRIPTION
**Pull request by jroesch**
I started messing around with Arvo and wrote a main in Rust so we can build and link the project via Cargo. I'll probably layer some more commits on here but figured we could hack and chat on this PR.

### CODE REVIEW

File: vernac.h [hunk 1]
```diff
@@ -37,6 +37,11 @@ command *make_data(variable* name, int num_constructors, int num_params);
 command *make_axiom(variable* name, term* ty);
 command *make_import(variable* name);
 
+// A helper that makes dealing with input streams easier from Rust.
```
Comment (jroesch) [inline vernac.h:40]: Rust has a cross-platform (windoze, osx, linux) abstraction for files and input streams and translating from Rust `Stdin` to a `FILE*` seemed more complicated then it was worth.

**Comment by jroesch** [comment]
Just did this before bed, was hoping to clean up the pretty printer, but if you are interested here are the formatting interfaces I was talking about:
- http://doc.rust-lang.org/stable/std/fmt/trait.Display.html
- http://doc.rust-lang.org/stable/std/fmt/struct.Formatter.html

The `Formatter` type both takes care of being efficient in terms of allocation, and enables common operations like padding and such.

**Comment by wilcoxjay** [comment]
Wow, awesome! I'm pretty excited about all this.

One thing is that I can't seem to build it using stable rust (rustc 1.2.0 (082e47636 2015-08-03)). Should we just move to unstable or whatever? How do I do that?

File: src/bin/arvo.rs [hunk 2]
```diff
@@ -0,0 +1,61 @@
+#![feature(convert)]
```
Comment (wilcoxjay) [inline src/bin/arvo.rs:1]: where do you use this?

**Comment by wilcoxjay** [comment]
Okay I installed nightly on my machine and confirmed that this works. It would be useful to make this PR on a branch that we can both push to. I've given you write access to this repository (via adding you to the uwplse group). Can you just push your rust-ify branch here? 

I think we may have to close this and open a new PR because everything is horrible, but maybe there's a way to change the branch.

File: src/bin/arvo.rs (hunk 2 above)
Comment (jroesch) [inline src/bin/arvo.rs:1]: This was just to get access to the `to_cstring()` conversion helper, we can just manually construct a `CString` if we want to remain on stable.

**Comment by jroesch** [comment]
I'm not sure if you are using it but I recommend managing Rust/Cargo install via [multirust](https://github.com/brson/multirust.git). It makes dealing with juggling stable and nightly easy, plus updating the nightly toolchain only requires a single command.

**Comment by jroesch** [comment]
I opened another PR this time from a branch on the main repository. It can be found here: https://github.com/uwplse/arvo/pull/15.

**Pull request closed without merging**